stardis/radiation_field/opacities/opacities_solvers/base.py

```diff
@@ -448,29 +443,116 @@ def calc_alpha_line_at_nu(
                 "Broadening range must be in units of length or frequency."
             )
 
-        # Iterate through each frequency to calculate the contribution of each line within the broadening range.
-        for i, nu in enumerate(tracing_nus):
-            delta_nus = nu.value - line_nus
+    # Iterate through each frequency to calculate the contribution of each line within the broadening range.
+    if parallel_config:
+        alpha_line_at_nu = calc_alan_entries_parallel(
+            stellar_model.no_of_depth_points,
+            tracing_nus.value,
+            delta_nus,
+            doppler_widths,
+            gammas,
+            alphas_array,
+            line_range_value,
+            h_lines_indices,
+        )
+
+    else:
+        alpha_line_at_nu = calc_alan_entries(
+            stellar_model.no_of_depth_points,
+            tracing_nus.value,
+            delta_nus,
+            doppler_widths,
+            gammas,
+            alphas_array,
+            line_range_value,
+            h_lines_indices,
+        )
+
+    return alpha_line_at_nu, gammas, doppler_widths
 
-            broadening_mask = np.abs(delta_nus) < line_range_value[i]
+
+@numba.njit(parallel=True)
+def calc_alan_entries_parallel(
+    no_of_depth_points,
+    tracing_nus_values,
+    delta_nus,
+    doppler_widths,
+    gammas,
+    alphas_array,
+    broadening_range=None,
+    h_lines_indices=None,
+):
+    alpha_line_at_nu = np.zeros((no_of_depth_points, len(tracing_nus_values)))
+
+    if broadening_range is None:
+        for frequency_index in numba.prange(len(tracing_nus_values)):
```

Comment:
Is there any configuration of the number of threads allowed for Numba?

Reply:
Yup. You can do numba.set_num_threads(int) to tell numba how many it is allowed to use. I think by default it tries to use all of them, but that's not guaranteed and you can see how many with numba.config.NUMBA_NUM_THREADS.

Reply:
I was thinking about letting the user set the max number of threads in the config, but I don't know if that's better than just letting them interact with numba directly. If we stick with the latter case, then we can probably just detail how to do that in a parallelization tutorial notebook that could be thrown together.

Reply:
I was thinking something user-configurable like TARDIS. Depends entirely on how accessible you want parallelization to be.

Reply:
That makes sense and I've updated it to behave that way. You specify a number of threads in the config. -99 (default) is serial, 0 is max number of threads, and any positive integer will be passed to numba. There's also a logger.info message stating what happens based on that number.

Reply:
I changed this to 1 for serial, and -99 for max threads